ytgov/travel-authorization
# Standardize `destinations` table in Database

# Context

See https://github.com/icefoganalytics/travel-authorization/issues/30 and [Modeling, My Travel Requests, 2023-10-13](https://docs.google.com/document/d/1jpW3VokZa3v5Mv-Gf4Kfjve2WwTpK7X4cpep5-EROCE/edit?pli=1#heading=h.epyy9mt07rdp)

1. Rename the `destinations` table to the more standard name of `locations`
2. Fix all foreign key references to the table so that they are called `location_id`.
3. Add `created_at` and `update_at` standard timestamp columns

# Implementation

Migrate back-end references
Migrate front-end api calls to use the `web/src/apis/locations-api.js`
Did _not_ migrate all front-end variables, as that turned out to be too large a project.

I believe this scope of work reflects the best gain for future developers, without the requirement to QA the entire front-end (which has zero automated tests).
Note that many of the changes are from trimming whitespace, which is usually don't by the developer's text editor.

# Testing Instructions

1. Boot the db and api services via `dev up`.
2. Boot the web service by `dev up_web`.
3. Check that the app boots correctly.
5. Go to the My Travel Request page.
6. Create a new travel request.
7. Check that the locations load in the "Form" and "To" selectors.
8. Check that you can save the form.
